pkg/controller/controller.go

```diff
@@ -298,6 +313,19 @@ func (c *Controller) renderBadRequest(w http.ResponseWriter, m string) {
 	})
 }
 
+func getEmail(iapEmailValue string) (string, error) {
```

Comment:
Nit: I would expect this to accept an `r *http.Request` instead. Shouldn't change much about the unit testing, but it provides a stronger encapsulation.